apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/create/components/client/create-ecosystem-form.client.tsx

```diff
@@ -0,0 +1,183 @@
+                      }}
+                    />
+                  </FormControl>
+                  <FormDescription>
+                    The logo will be displayed as the ecosystem's wallet icon.
+                  </FormDescription>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+            <FormField
+              control={form.control}
+              name="permissions"
+              render={({ field }) => (
+                <FormItem>
+                  <FormLabel>Integration permissions</FormLabel>
+                  <FormControl>
+                    <RadioGroup
+                      onValueChange={field.onChange}
+                      defaultValue="PARTNER_WHITELIST"
+                      className="flex gap-4 py-2"
+                    >
+                      <FormItem>
+                        <FormControl>
+                          <RadioGroupItem
+                            value="PARTNER_WHITELIST"
+                            id="PARTNER_WHITELIST"
+                          >
+                            Allowlist
+                          </RadioGroupItem>
+                        </FormControl>
+                      </FormItem>
+                      <FormItem>
+                        <FormControl>
+                          <RadioGroupItem value="ANYONE" id="ANYONE">
+                            Public
+                          </RadioGroupItem>
+                        </FormControl>
+                      </FormItem>
+                    </RadioGroup>
+                  </FormControl>
+                  <FormDescription>
+                    <Link
+                      className="text-primary"
+                      href="https://portal.thirdweb.com/connect/ecosystems/ecosystem-permissions"
+                    >
+                      Learn more
+                    </Link>{" "}
+                    about ecosystem permissions
+                  </FormDescription>
+                  <FormMessage />
+                </FormItem>
+              )}
+            />
+          </div>
+          <Button
+            type="submit"
+            variant="primary"
+            className="min-w-28"
+            disabled={isLoading}
+          >
+            {isLoading && <Loader2 className="w-4 h-4 mr-2 animate-spin" />}
+            Create
+          </Button>
+        </form>
+      </Form>
+      <ConfirmationDialog
+        open={confirmationOpen}
+        onOpenChange={setConfirmationOpen}
+        title={`Are you sure you want to create ecosystem ${nameToSlug(
+          form.getValues().name,
+        )}?`}
+        description="Your account will be charged $250 per month."
+        onSubmit={() =>
+          createEcosystem({
+            id: nameToSlug(form.getValues().name),
+            name: form.getValues().name,
+            logo: form.getValues().logo,
+            permission: form.getValues().permissions,
+          })
```

Comment:
related to the comment above, probably just grab all this data out of the state, or *if* you need to pull it out of the form here then at least only call the getValues once and access the resultiong object :)